File: CoorpoSanpedroBackend/src/main/java/com/festival/entity/TipoCuentaBancaria.java
package com.festival.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

/**
 * Catálogo de tipos de cuenta bancaria.
 * Reemplaza al enum TipoCuentaBancaria para permitir gestión dinámica.
 */
@Entity
@Table(name = "tipos_cuenta_bancaria")
@Getter
@Setter
@NoArgsConstructor
public class TipoCuentaBancaria extends BaseEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, unique = true, length = 100)
    private String nombre;

    @Column(length = 255)
    private String descripcion;

    @Column(nullable = false)
    private boolean activo = true;

    public TipoCuentaBancaria(String nombre) {
        this.nombre = nombre;
    }
}

File: frontend/src/models/tipoCuentaBancaria.ts
export interface TipoCuentaBancaria {
  id: number;
  nombre: string;
  descripcion?: string;
  activo: boolean;
}

export interface CreateTipoCuentaDTO {
  nombre: string;
  descripcion?: string;
}

export interface UpdateTipoCuentaDTO {
  nombre: string;
  descripcion?: string;
}
